### PR TITLE
Better API

### PR DESCRIPTION
**Pull request by levydsa**
Depends on #13

### CODE REVIEW

File: .kotlin/errors/errors-1723820326435.log [hunk 1]
```diff
@@ -0,0 +1,4 @@
+kotlin version: 2.0.10
```
Comment (haaawk) [inline .kotlin/errors/errors-1723820326435.log:1]: Do we really want to commit this?

File: libsql/src/main/kotlin/tech/turso/libsql/Transaction.kt [hunk 2]
```diff
@@ -57,10 +85,24 @@ class Transaction internal constructor(private var inner: Long) : Connection {
...
+                is Int -> Value.newBuilder().setInteger(value.toLong()).build()
+                is Long -> Value.newBuilder().setInteger(value).build()
+                is String -> Value.newBuilder().setText(value).build()
+                is Float -> Value.newBuilder().setReal(value.toDouble()).build()
+                is Double-> Value.newBuilder().setReal(value).build()
+                null -> Value.newBuilder().setNull(Value.Null.newBuilder().build()).build()
+                else -> {
+                    throw IllegalArgumentException("Type not supported")
+                }
+            }
```
Comment (haaawk) [inline libsql/src/main/kotlin/tech/turso/libsql/Transaction.kt:103]: This is being duplicated at least 8 times. Worth having a helper method?

**Review by haaawk — COMMENTED** [review]
LGTM

File: libsql/src/main/kotlin/tech/turso/libsql/Transaction.kt (hunk 2 above)
Comment (levydsa) [reply libsql/src/main/kotlin/tech/turso/libsql/Transaction.kt:103]: Yes, I'll do that

**Review by haaawk — APPROVED** [review]

**Comment by haaawk** [comment]
Rebased and merged here -> https://github.com/tursodatabase/libsql_android/pull/16

**Pull request closed without merging**